receipt_dynamo/receipt_dynamo/data/base_operations/flattened_mixin.py

```
@@ -117,9 +137,98 @@ def _validate_pagination_params(
+        Raises:
+            EntityValidationError: If keys are invalid
+        """
+        if keys is None:
+            raise EntityValidationError("keys cannot be None")
+        if not isinstance(keys, list):
+            raise EntityValidationError("keys must be a list")
+        if not all(isinstance(key, dict) for key in keys):
+            raise EntityValidationError("keys must be a list of dictionaries")
+        for key in keys:
+            if not {"PK", "SK"}.issubset(key.keys()):
+                raise EntityValidationError("keys must contain 'PK' and 'SK'")
```

Ruff (0.14.13)

Avoid specifying long messages outside the exception class

(TRY003)